modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb

```diff
@@ -39,15 +39,20 @@ def sort(facilities, sort_method)
 
       def sort_by_recent_appointment(facilities)
         facilities.sort_by(&:name) # entries not in recent appointments will be sorted alphabetically
-        appointments = Mobile::V0::Appointment.get_cached(@current_user).sort_by(&:start_date_utc)
-        if appointments.blank?
-          raise Common::Exceptions::RecordNotFound.new(
-            detail: 'Could not fetch user appointments', source: self.class.to_s
-          )
+        appointments = Mobile::V0::Appointment.get_cached(@current_user)&.sort_by(&:start_date_utc)
+        if appointments.nil?
+          Rails.logger.info('mobile facilities info appointments cache nil', user_uuid: @current_user.uuid)
+          facilities
+        else
+          recent_facilities = appointments.map(&:facility_id).uniq
+
+          recent_facilities = recent_facilities.map do |facility|
+            Mobile::V0::Appointment.convert_to_non_prod_id!(facility)
+          end
+
+          rf_hash = recent_facilities.each_with_index.to_h
+          facilities.sort_by { |facility| [rf_hash[facility.id] || recent_facilities.size, facility.id] }
```

Comment:
If we do this in the case of an empty appointments, we just end up sorting facilities by id. Based on our conversation in slack, I was expecting us to sort alphabetically but not log. So more like
```
if appointments.nil?
  # log
  facilities
elsif appointments.empty?
  facilities
else
  ...
end
```

I think the logic here is basically "sort the user's facilities by recency of appointment and if they haven't had an appointment at the facility just tack it on at the end."

Reply:
This is correct. Those tacked on facilities should be sorted alphabetically

Reply:
I believe the change to `facilities.sort_by(&:name)` should mean `facilities` is sorted alphabetically right from the beginning. Then code in the `else` block should just return facilities sorted alphabetically because it already returns "leftover" facilities alphabetically. Unless you like the `elsif appointements.empty?` for code clarity

Reply:
That was the intended functionality. @Tonksthebear just write or make sure the existing tests are adequately ensuring that's what's happening